core/embed/rust/src/micropython/macros.rs

```diff
@@ -164,3 +170,23 @@ macro_rules! obj_type {
         }
     }};
 }
+
+/// Construct an extmod definition.
+macro_rules! obj_module {
+    ($globals:expr) => {{
+        #[allow(unused_unsafe)]
+        unsafe {
+            use $crate::micropython::ffi;
+
+            ffi::mp_obj_module_t {
+                base: ffi::mp_obj_base_t {
+                    type_: &ffi::mp_type_module,
+                },
+                // TODO: This is safe only if we pass in `Dict` with fixed `Map` (created by
+                // `Map::fixed()`, usually through `obj_map!`), because only then will
+                // MicroPython treat `globals` as immutable, and make the mutable cast safe.
```

Comment:
Would it be possible/difficult/ugly to just call the `obj_dict!(obj_map!(...))` from here instead?

Reply:
Not difficult, just ugly, we'd need to copy the field matchers from the macro definitions into this one. But I guess we should. Or we can maybe make a newtype for the static dicts, but I don't like it too much.

Reply:
Copied the field matchers in [681ac69](https://github.com/trezor/trezor-firmware/pull/2138/commits/681ac69b3f12cbb842e46f646cdd2047b6bcc00b).

Reply:
the comment needs to be tweaked: "This is safe only because we are passing in a static dict with a fixed Map created by (...). That causes MicroPython to treat `globals` as immutable (...)"

Reply:
744f4d6bfafbc0b7676ff3bdead8160af6441bba